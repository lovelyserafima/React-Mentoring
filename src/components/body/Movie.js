import React from 'react';
import MovieTileWrapper from "../../style/body/Movie.Styles";
import MoviePoster from "./MoviePoster";
import MovieInfo from "./MovieInfo";
import MovieGenre from "./MovieGenre";

const Movie = ({ data: { poster_path, title, release_date, genres } }) => (
    <MovieTileWrapper>
        <MoviePoster posterImage={poster_path} />
        <MovieInfo title={title} release_date={release_date} />
        <MovieGenre genres={genres} />
    </MovieTileWrapper>
);

export default Movie;