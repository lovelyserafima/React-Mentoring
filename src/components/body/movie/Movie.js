import React from 'react';
import MovieTileWrapper from "./Movie.Styles";
import MoviePoster from "../movieposter/MoviePoster";
import MovieInfo from "../movieinfo/MovieInfo";
import MovieGenre from "../moviegenre/MovieGenre";
import { connect } from 'react-redux';
import {selectMovie} from "../../../redux/actions";

export const Movie = ({
  data,
  data: { poster_path, title, release_date, genres },
  selectMovie
}) => (
    <MovieTileWrapper onClick={() => {selectMovie(data)}}>
        <MoviePoster posterImage={poster_path} />
        <MovieInfo title={title} release_date={release_date} />
        <MovieGenre genres={genres} />
    </MovieTileWrapper>
);

export default connect(
    null,
    { selectMovie }
)(Movie);