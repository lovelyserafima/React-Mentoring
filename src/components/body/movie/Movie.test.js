import React from 'react';
import { shallow } from 'enzyme';

import Movie from './Movie';

describe('MovieTile', () => {
    const data = {
        poster_path:
            'https://image.tmdb.org/t/p/w500/1Ilv6ryHUv6rt9zIsbSEJUmmbEi.jpg',
        release_date: '2013-04-18',
        title: 'Iron Man 3',
        genres: ['Action', 'Adventure', 'Science Fiction']
    };
    it('should render correctly', () => {
        const component = shallow(<Movie data={data} />);

        expect(component).toMatchSnapshot();
    });
});