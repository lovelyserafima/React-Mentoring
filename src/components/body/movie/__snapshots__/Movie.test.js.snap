// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile should render correctly 1`] = `ShallowWrapper {}`;
